electricitymap/contrib/libs/models/datapoints.py

```diff
@@ -0,0 +1,427 @@
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta, timezone
+from logging import Logger
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, validator
+
+from electricitymap.contrib.config import EXCHANGES_CONFIG, ZONES_CONFIG, ZoneKey
+from electricitymap.contrib.libs.models.constants import VALID_CURRENCIES
+
+LOWER_DATETIME_BOUND = datetime(2000, 1, 1, tzinfo=timezone.utc)
+
+
+class ProductionMix(BaseModel):
+    """
+    Containts the production mix for a zone at a given time.
+    All values are in MW.
+    """
+
+    biomass: Optional[float] = None
+    coal: Optional[float] = None
+    gas: Optional[float] = None
+    geothermal: Optional[float] = None
+    hydro: Optional[float] = None
+    nuclear: Optional[float] = None
+    oil: Optional[float] = None
+    solar: Optional[float] = None
+    unknown: Optional[float] = None
+    wind: Optional[float] = None
+
+    def set_value(self, mode: str, value: float) -> None:
+        """
+        Sets the value of a production mode.
+        This can be used if the Production has been initialized empty
+        and is being filled in a loop.
+        """
+        self.__setattr__(mode, value)
+
+
+class StorageMix(BaseModel):
+    """
+    Containts the storage mix for a zone at a given time.
+    All values are in MW.
+    """
+
+    battery: Optional[float] = None
+    hydro: Optional[float] = None
```

Comment:
This should also have a set_value function as it will most likely be used in tandem with ProductionMix if there is storage.

It will also make sure our "API" surface is the same between the two.